sopel/modules/emailcheck.py

```diff
@@ -0,0 +1,314 @@
+    gline_time = ValidatedAttribute('gline_time', default="24h")
+    exempt_suffixes = ListAttribute("exempt_suffixes")
+    warn_chans = ListAttribute("warn_chans")
+    protect_chans = ListAttribute("protect_chans")
+
+def configure(config):
+    config.define_section('emailcheck', EmailCheckSection)
+    config.emailcheck.configure_setting('IPQS_key',
+                                        'Access key for IPQS service')
+    config.emailcheck.configure_setting('disallow_threshold',
+                                        'Addresses with scores >= this will be disallowed; no punishment')
+    config.emailcheck.configure_setting('malicious_threshold',
+                                        'Addresses with scores >= this will be interpreted as attacks')
+    config.emailcheck.configure_setting('gline_time',
+                                        'Users attempting to register with malicious addresses will be '
+                                        'glined for this priod of time.')
+    config.emailcheck.configure_setting('exempt_suffixes',
+                                        'Suffixes (TLD, whole domain, etc.) to exempt from checking')
+    config.emailcheck.configure_setting('warn_chans',
+                                        'List of channels to warn when a suspicious user is detected. '
+                                        'May be empty.')
+    config.emailcheck.configure_setting('protect_chans',
+                                        'List of channels to +R after malicious attempt to reg. '
+                                        'May be empty.')
+
+def setup(bot):
+    bot.config.define_section('emailcheck', EmailCheckSection)
+
+@dataclass
+class Email:
+    user: str
+    domain: str
+    def get_address(self):
+        return f'{self.user}@{self.domain}'
+    def __str__(self):
+        return self.get_address()
+    def __post_init__(self):
+        self.domain = self.domain.lower()
+
+@dataclass
+class DomainInfo:
+    score: float
+    flag_disposable: bool
+    flag_recent_abuse: bool
+
+def alert(bot, alert_msg: str, log_err: bool = False):
+    for channel in bot.config.emailcheck.warn_chans:
+        bot.say(alert_msg, channel)
+    if log_err:
+        LOGGER.error(alert_msg)
+
+def add_badmail(bot, email):
+    #Right now we're BADMAILing whole domains. This might change.
+    if safe_mode:
+        LOGGER.info(f"SAFE MODE: Would badmail {email}")
+    else:
+        bot.write("NICKSERV", "badmail", "add", f'*@{email.domain}')
+
+def fdrop(bot, nick: str):
+    if safe_mode:
+        LOGGER.info(f"SAFE MODE: Would fdrop {nick}")
+    else:
+        bot.write("NICKSERV", "fdrop", nick.lower())
+
+def gline_ip(bot, ip: str, duration: str):
+    if safe_mode:
+        LOGGER.info(f"SAFE MODE: Would gline {ip} for {duration}")
+    else:
+        bot.write("GLINE", f'*@{ip}', duration, KILL_STR)
+
+def gline_username(bot, nick: str, duration: str):
```

Comment:
Is this only for irccloud? If so I think the function should be named as such, like `gline_irccloud` or such.

Reply:
I went back and forth on that, heh